### ISSUE TITLE
Cannot pair

### ISSUE THREAD

**Issue by j4k3** [HomeACcessoryKid/ESP8266-HomeKit#28]
I've got everything to run up to the point where my ESP8622 with ESP8622-HomeKit-Demo is discoverable in the Home app. The app then asks me for the pairing code, which for me still is the default one.
However, the procss doesn't finish after 25 seconds for me as the tutorial states. It instead times out after 120 seconds, when the app tells me it cannot connect to the device. I tried to increase the clock speed to 80MHz in the build process, thinking that would help speeding up the crypto but to no avail.

Is there anything I can do to debug this process? How do I compile the code with debugging on?
I am, by the way on the current iOS 11 beta with all my devices, is anyone else experiencing problems on these?

**Comment by HomeACcessoryKid** [comment]
You must manage to run a hello-world on your device and get serial reporting in place
My code delivers tons of debug info and without it no-one can help you.
You actually must change something to switch it off.

Meanwhile, the other day I could not pair anymore, and found that the only way to get it working again was to delete my ‘home’ in the Home-app on the phone.
Maybe this applies to you as well, but you MUST get serial working, else there is no point in working with this software...

**Comment by j4k3** [comment]
Now thats weird. I've just tried to connect again after leaving the controller powered up yesterday. It worked like a charm. It just worked as if I didn't try and failed at doing the same thing, a dozen times from two different devices yesterday.

**Comment by HomeACcessoryKid** [comment]
Good to hear that you got success.
please open a new issue with this new question, to keep things clean

**Comment by HomeACcessoryKid** [comment]
Actually, put that as an issue on the Demo page. Myself I have not too much time to test out this kind of stuff (alas) but others might share their experiences.

**Comment by abhinavjha86** [comment]
Hi,

I managed to pair the demo accessory only once, since then no matter what I do, I always get ''.
I tried erasing the flash, resetting HomeKit configuration on my iPhone but nothing is working.
I am attaching the logs.... please help.
[logs.txt](https://github.com/HomeACcessoryKid/ESP8266-HomeKit/files/1253931/logs.txt)

**Comment by HomeACcessoryKid** [comment]
Hi, I think I had the same issue myself (and wondering if I went crazy).
My 'solution' in the end was to delete the 'Home' and start over. Then it worked.
Did you sync recently? I have added the fix to have a random Username (looks like a MAC) which is supposed to be according to spec. That could also help...

Good luck and report back?

**Comment by abhinavjha86** [comment]
I cloned you repository today itself and the random username generation is working fine every time I erase the flash... also I delete 'home' from home app but still getting the 'client 192.168.0.4:49242 disconnected with status 0' issue... don't know why only I am facing it... is there a way to know exact reason of disconnection from the SDK API calls ??

**Comment by HomeACcessoryKid** [comment]
Since it is the iPhone disconnecting we have to ask Apple… I did not see an error in the log
Sometimes life is hard, fighting your way through will learn you a lot. But no guarantees I’m afraid.

**Comment by abhinavjha86** [comment]
true... I'll keep trying,, anyways thanks for this great effort of yours,, looking forward for the Arduino library... wish I could also contribute but I am limited by my knowledge ;-)

**Comment by Andreaqc** [comment]
@abhinavjha86 it seems like you might be losing packets, but i am not so sure about it.
Could you try again with my [branch](https://github.com/Andreaqc/ESP8266-HomeKit) and then attach the output with DEBUG0,1,2,3,4 defined?

**Comment by HomeACcessoryKid** [comment]
@abhinavjha86 it might indeed be packets that are lost. My home wifi is also not OK and I see disconnects all the time. This is not something I consider a problem since it will be part of life of many people. The thing is that even though this happens, the long term stability is not jeopardised. Actually, the last few months of the original repository were spend on making everything robust, explaining some of the more obscure parts of the code.

Can you confirm that in itself your device works well (part of the time) or you have use cases that NEVER work? My testcase is setting the 'light' to 50% and see if a week later it still is there. Like that you can know if it has reset, bringing the 'light' back to 0%.

**Comment by HomeACcessoryKid** [comment]
IMPORTANT  
When working on the RGBW lamp I have found "THE BUG" that must have made pairing a matter of luck for many. The change I have pushed one hour ago solves this issue where packets were ignored, and especially in the case of pairing. It has to do with timing and the cause was a delay that I introduced myself in crypto_tasks... I recommend everyone to sync up to at least this version.

At the same time there is some dark magic that whenever syncing was not clean, deleting the 'home' seems to be the only remedy... 

PS. I will investigate how to fix the WPA2 vulnerability... ESP8266_RTOS has changed some stuff in a not backward compatible way so I will go to a separate branch I guess.
